### ISSUE TITLE
send_query(): listfields() crashes if run after getresult() returns None

### ISSUE THREAD

**Issue by justinpryzby** [PyGreSQL/PyGreSQL#80]
```python
import pg
db=pg.DB('postgres')
res=db.send_query('SELECT 1')
res.getresult()
res.getresult()
res.listfields()
```

Results in:
```
Violación de segmento (`core' generado)
```

Maybe another buglet: the reference to the "db" object seems to be lost before it should:
```python
import pg
db=pg.DB('postgres')
res=db.send_query('SELECT 1')
db=None
res.getresult()
res.getresult()
```

Results in:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: Connection is not valid
```

**Comment by Cito** [comment]
This is clearly triggered by the second call to `getresult()` and how the case that no result is returned is handled [here](https://github.com/PyGreSQL/PyGreSQL/blob/373e1a93ca33a8b4254772bbddf360ce989d907c/pgquery.c#L142).

I suggest that instead of 
```c
Py_DECREF(self->pgcnx);
self->pgcnx = NULL;
```
we just set
```c
self->max_row = 0;
self->num_fields = 0;
self->col_types = NULL;
```
Do you think that would work as a fix?

**Comment by Cito** [comment]
Fixed in 53e8f10d2eb6d2ab4179ec4c8fee4495cbd9852d.
